torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -182,6 +183,9 @@ def __init__(
         super().__init__()
 
         assert len(configs) > 0, "Non-empty TritonConfig list required for compiling"
+        # makes sure there are no pre-hooks on any of the triton configs
+        (validate_triton_config(cfg) for cfg in configs)
```

Comment:
not sure if there's a better place to put the check? I chose here because the CachingAutotuner is the place where the triton runtime behavior is modeled (and where the pre_hook behavior is being omitted now), but open to suggestions if there's another place that's more obvious